.drone.star

```diff
@@ -0,0 +1,258 @@
+def main(ctx):
+  before = [
+      testing(ctx),
+  ]
+
+  stages = [
+      changelog(ctx),
+      build(ctx),
+  ]
+
+  return before + stages
+
+def testing(ctx):
+    return {
+        'kind': 'pipeline',
+        'type': 'docker',
+        'name': 'testing',
+        'platform': {
+            'os': 'linux',
+            'arch': 'amd64',
+        },
+        'steps': [
+            {
+                'name': 'lint',
+                'image': 'webhippie/nodejs:latest',
+                'pull': 'always',
+                'commands': [
+                    'yarn install',
+                    'yarn lint',
+                ]
+            },
+        ],
+        'trigger': {
+            'ref': [
+                'refs/heads/master',
+                'refs/tags/**',
+                'refs/pull/**',
+            ],
+        },
+    }
+
+def build(ctx):
+    return {
+        'kind': 'pipeline',
+        'type': 'docker',
+        'name': 'build',
+        'platform': {
+            'os': 'linux',
+            'arch': 'amd64',
+        },
+        'steps': [
+            {
+                'name': 'build-docs',
+                'image': 'webhippie/nodejs:latest',
+                'pull': 'always',
+                'commands': [
+                    'yarn install',
+                    'yarn build:docs',
+                ]
+            },
+            {
+                'name': 'build-system',
+                'image': 'webhippie/nodejs:latest',
+                'pull': 'always',
+                'commands': [
+                    'yarn install',
+                    'yarn build:system',
+                ]
+            },
+            {
+                'name': 'publish-docs',
+                'image': 'plugins/gh-pages:1',
+                'pull': 'always',
+                'settings': {
+                    'username': {
+                        'from_secret': 'github_username',
+                    },
+                    'password': {
+                        'from_secret': 'github_token',
+                    },
+                },
+                'pages_directory': 'dist/docs',
+                'when': {
+                    'ref': {
+                        'exclude': [
+                            'refs/pull/**',
+                        ],
+                    },
+                },
+            },
+            {
+                'name': 'publish-system',
+                'image': 'plugins/npm:1',
+                'pull': 'always',
+                'settings': {
+                    'username': {
+                        'from_secret': 'npm_username',
+                    },
+                    'email': {
+                        'from_secret': 'npm_email',
+                    },
+                    'token': {
+                        'from_secret': 'npm_token',
+                    },
+                },
+                'when': {
+                    'ref': {
+                        'exclude': [
```

Comment:
Why not use also tags as a ref instead of excluding as done in changelog and release?

Reply:
done